src/OpenCV/ImgProc/MiscImgTransform.hsc

```diff
@@ -1167,11 +1167,49 @@ grayscale image or for removing a noise, that is, filtering out pixels with
 too small or too large values. There are several types of thresholding
 supported by the function.
 
+Example:
+
+@
+grayBirds :: Mat (ShapeT [341, 512]) ('S 1) ('S Word8)
+grayBirds = exceptError $ cvtColor bgr gray birds_512x341
+
+threshBinaryBirds :: Mat (ShapeT [341, 512]) ('S 3) ('S Word8)
+threshBinaryBirds =
+    exceptError $ cvtColor gray bgr $ fst $ exceptError $
+    threshold (ThreshVal_Abs 100) (Thresh_Binary 150) grayBirds
+    
+threshBinaryInvBirds :: Mat (ShapeT [341, 512]) ('S 3) ('S Word8)
+threshBinaryInvBirds =
+    exceptError $ cvtColor gray bgr $ fst $ exceptError $
+    threshold (ThreshVal_Abs 100) (Thresh_BinaryInv 150) grayBirds
+    
+threshTruncateBirds :: Mat (ShapeT [341, 512]) ('S 3) ('S Word8)
+threshTruncateBirds =
+    exceptError $ cvtColor gray bgr $ fst $ exceptError $
+    threshold (ThreshVal_Abs 100) Thresh_Truncate grayBirds
+    
+threshToZeroBirds :: Mat (ShapeT [341, 512]) ('S 3) ('S Word8)
+threshToZeroBirds =
+    exceptError $ cvtColor gray bgr $ fst $ exceptError $
+    threshold (ThreshVal_Abs 100) Thresh_ToZero grayBirds
+    
+threshToZeroInvBirds :: Mat (ShapeT [341, 512]) ('S 3) ('S Word8)
+threshToZeroInvBirds =
+    exceptError $ cvtColor gray bgr $ fst $ exceptError $
+    threshold (ThreshVal_Abs 100) Thresh_ToZeroInv grayBirds
+@
+
+<<doc/generated/examples/threshBinaryBirds.png threshBinaryBirds>>
+<<doc/generated/examples/threshBinaryInvBirds.png threshBinaryInvBirds>>
+<<doc/generated/examples/threshTruncateBirds.png threshTruncateBirds>>
+<<doc/generated/examples/threshToZeroBirds.png threshToZeroBirds>>
+<<doc/generated/examples/threshToZeroInvBirds.png threshToZeroInvBirds>>
+
 <http://docs.opencv.org/3.0-last-rst/modules/imgproc/doc/miscellaneous_transformations.html#threshold OpenCV Sphinx doc>
 -}
 threshold
     :: (depth `In` [Word8, Float])
-    => ThreshValue -- ^
+    => ThreshValue
```

Comment:
We usually put in an empty `-- ^` to ensure haddock will put the individual arguments on separate lines which is easier to read. Lets leave it in.